pkg/config/validation/agent/validation.go

```diff
@@ -673,6 +676,50 @@ func ValidateWildcardDomain(domain string) error {
 	return nil
 }
 
+// ValidateWildcardDomainForVirtualServiceBoundToGateway checks that a domain is a valid FQDN, but also allows wildcard prefixes.
+// If it is an SNI domain, then it does a special validation where it allows a
+// which matches outbound_.<PORT>_._.<HOSTNAME> format
+func ValidateWildcardDomainForVirtualServiceBoundToGateway(sni bool, domain string) error {
+	if err := CheckDNS1123Preconditions(domain); err != nil {
+		return err
+	}
+
+	// check if its an auto generated domain, with outbound_ as a prefix.
+	if sni && strings.HasPrefix(domain, outboundHostPrefix) {
+		// example of a validate domain: outbound_.80_._.e2e.foobar.mesh
+		trafficDirectionSuffix, port, hostname, err := parseAutoGeneratedSNIDomain(domain)
+		if err != nil {
+			return err
+		}
+		if trafficDirectionSuffix != outboundHostPrefix {
+			return fmt.Errorf("domain name %q invalid (label %q invalid)", domain, trafficDirectionSuffix)
+		}
+		match, _ := regexp.MatchString("([0-9].*)", port)
+		if !match {
+			return fmt.Errorf("domain name %q invalid (label %q invalid). should follow %s format", domain, port, outboundHostNameFormat)
+		}
+		match, _ = regexp.MatchString("([a-zA-Z].*)", hostname)
+		if !match {
+			return fmt.Errorf("domain name %q invalid (label %q invalid). should follow %s format", domain, hostname, outboundHostNameFormat)
+		}
+		return nil
+	}
+	return ValidateWildcardDomain(domain)
+}
+
+// parseAutoGeneratedSNIDomain parses auto generated sni domains
+// which are generated when using AUTO_PASSTHROUGH mode in envoy
+func parseAutoGeneratedSNIDomain(domain string) (string, string, string, error) {
```

Comment:
Not a big deal to re-implement it since its fairly simple, but is there a reason we are not using model.ParseSubsetKey directly?

Reply:
oh my bad. I didn't know this function existed. Thought you wrote sudo code. Let me check.

Reply:
There is an import cycle issue with ParseSubsetKey. What do you recommend? Copy the function to some utility package?